source/h1_decoder.c

```diff
@@ -514,6 +523,15 @@ static int s_linestate_header(struct aws_h1_decoder *decoder, struct aws_byte_cu
                 return aws_raise_error(AWS_ERROR_HTTP_PARSE);
             }
 
+            if (decoder->body_headers_forbidden) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_HTTP_STREAM,
+                    "id=%p: Incoming headers for content-length received, but incoming status forbids it. This is "
+                    "illegal.",
```

Comment:
nitpick: Keep the message vague,  since we might find more reasons in the future to forbid a body.
"Incoming headers for content-length received, but it is illegal for this message to have a body."